plugins/synapticsmst/synapticsmst-device.c

```diff
@@ -504,32 +541,502 @@ synapticsmst_device_get_flash_checksum (SynapticsMSTDevice *device,
+	tmpCRC = synapticsmst_device_get_crc( 0, 16, fw_size, payload_data );
+	tagData[0] = BankToUpdate;
+	tagData[4] = (tmpCRC >> 8) & 0xFF;
+	tagData[5] = tmpCRC & 0xFF;
+	tagData[15] = (guint8)synapticsmst_device_get_crc(0, 8, 15, tagData);
+	#if MSTDBG
+	g_debug("tag date %x %x %x",tagData[1],tagData[2], tagData[3]);
+	g_debug("tag crc %x %x %x %x",tagData[0],tagData[4], tagData[5], tagData[15]);
+	#endif
+	do{
+		rc = 0;
+		rc = synapticsmst_common_rc_set_command (connection,
+								 UPDC_WRITE_TO_EEPROM, 16, (EEPROM_BANK_OFFSET * BankToUpdate + EEPROM_TAG_OFFSET), tagData);
+		if (rc) {
+			g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_INVALID_DATA,
+						"Failed to write TAG");
+		}
+		g_usleep(200);
+		rc = synapticsmst_common_rc_get_command (connection,
+								UPDC_READ_FROM_EEPROM, 16, (EEPROM_BANK_OFFSET * BankToUpdate + EEPROM_TAG_OFFSET), readBuf);
+		if (rc) {
+			g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_INVALID_DATA,
+						"Failed to read Tag");
+		}
+		else{
+			gboolean match = TRUE;
+			for (i=0; i<16; i++){
+				#if MSTDBG
+				g_debug("[%x]set tag vaild read %x tag %x", retryCount,readBuf[i], tagData[i]);
+				#endif
+				if (readBuf[i] != tagData[i]){
+					match = FALSE;
+				}
+			}
+			if (match){
+				break;
+			}
+		}
+	}while(retryCount++ < MAX_RETRY_COUNTS );
+	if( rc && retryCount== MAX_RETRY_COUNTS){
+				g_set_error_literal (error,	G_IO_ERROR,	G_IO_ERROR_INVALID_DATA,
+									"set tag fail");
+	}
+	/* set tag invaild*/
+	if(rc == 0){
+		rc = synapticsmst_common_rc_get_command (connection,
+							 UPDC_READ_FROM_EEPROM,
+							 1, (EEPROM_BANK_OFFSET * BankInUse + EEPROM_TAG_OFFSET + 15), tagData);
+		if (rc) {
+			g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_INVALID_DATA,
+					     "Failed to read from EEPROM of device");
+		}
+			
+		do
+		{
+			if (tagData[0] != 0xFF)	{  
+			/* CRC8 is not 0xFF, erase last 4k of bank# */
+				eraseOffset = (EEPROM_BANK_OFFSET * BankInUse + EEPROM_BANK_OFFSET - 0x1000) / 0x1000; // offset for last 4k of bank#
+				rc = synapticsmst_device_set_flash_sector_erase(device, FLASH_SECTOR_ERASE_4K, eraseOffset);				
+			}
+			else{  /* CRC8 is 0xFF, set it to 0x00 */
+				tagData[1] = 0x00;
+				rc = synapticsmst_common_rc_set_command (connection,
+								 UPDC_WRITE_TO_EEPROM,
+								 1, (EEPROM_BANK_OFFSET * BankInUse + EEPROM_TAG_OFFSET + 15), &tagData[1]);				
+			}
+
+			rc = synapticsmst_common_rc_get_command (connection,
+								UPDC_READ_FROM_EEPROM,
+								1, (EEPROM_BANK_OFFSET * BankInUse + EEPROM_TAG_OFFSET + 15), readBuf);
+			if (rc) {
+				g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_INVALID_DATA,
+							"Failed to read from EEPROM of device");
+			}
+			else{
+				if ((readBuf[0] == 0xFF && tagData[0] != 0xFF) || (readBuf[0] == 0x00 && tagData[0] == 0xFF))				{
+					break;
+				}
+			}
+		}while( retryCount++ < MAX_RETRY_COUNTS );
+
+		#if MSTDBG
+		g_debug("Panamera update done");		
+		#endif
+		return TRUE;
+	}
+	else{
```

Comment:
This else is redundant like this.

I'd say rather than pay an indentation for the normal case you can instead do something like:
```
if (rc != 0)
	return FALSE;
<all the rest of the stuff previously in if>
return TRUE;
```

Reply:
Sure